azure/scope/managedcontrolplane.go

```diff
@@ -511,6 +512,14 @@ func (s *ManagedControlPlaneScope) GetAgentPoolSpecs(ctx context.Context) ([]azu
 			ammp.Replicas = *ownerPool.Spec.Replicas
 		}
 
+		if ownerPool.Spec.Template.Spec.Version != nil {
```

Comment:
We should really have a UT for this (useful!) validation failure

Reply:
agreed, @zmalik would you be able to add a short UT for this? Sorry for not suggesting that earlier!

Reply:
sounds good! added the test now